text/trace/0226-sampling-random-traceids.md

```diff
@@ -0,0 +1,185 @@
+# Non-power-of-two Probability Sampling using 56 random TraceID bits
+
+## Motivation
+
+The existing, experimental [specification for probability sampling using TraceState](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-probability-sampling.md)
+supporting Span-to-Metrics pipelines is limited to powers-of-two
+probabilities and is designed to work without making assumptions about
+TraceID randomness.
+
+This proposes to extend that specification with support for 56-bit
+precision sampling probability.  This is seen as particularly
+important for implementation of probabilistic tail samplers (e.g., in
+the OpenTelemetry Collector) as explained below.
+
+This OTEP makes use of the [draft-standard W3C tracecontext `random`
+flag](https://w3c.github.io/trace-context/#random-trace-id-flag),
+which is an indicator that 7 bytes of true randomness are available
+for probability sampler decisions.
+
+## Explanation
+
+The existing, experimental TraceState probability sampling
+specification relies on two variables known as **r-value** and
+**p-value**.  The r-value carries the source of randomness and the
+p-value carries the effective sampling probability.
+
+Given this specification, a ConsistentProbabilitySampler can be
+applied as a head sampler for non-power-of-two sampling probabilities
+using interpolation.  For example, a neffective sampling probability
+of 1-in-3 can be achieved by alternating between 25% and 50% sampling.
+However, interpolation only works for trace roots, otherwise
+"consistent" sampling can only be achieved at the next smaller power
+of two.  In the example, sampling at 1-in-3 using interpolation means
+traces are only guaranteed **consistent** at 25% and smaller sampling
+probabilities.
+
+The major downside of the r-value, p-value approach is that r-value
+must be encoded even for unsampled contexts.  Ideally, building
```

Comment:
It would be nice to get this claim supported by some numbers. Assuming that the concern is about the wire protocol, it is true that passing the r-value will typically require a separate http header, but this should be seen in context of already necessary trace-id and parent span id which have to be there as well (for unsampled contexts). IMHO the overhead added by the r-value is fairly small and is outweighed by the benefits.